common/djangoapps/third_party_auth/samlproviderdata/views.py

```diff
@@ -52,15 +53,18 @@ def get_queryset(self):
             EnterpriseCustomerIdentityProvider,
             enterprise_customer__uuid=self.requested_enterprise_uuid
         )
-        saml_provider = SAMLProviderConfig.objects.get(pk=enterprise_customer_idp.provider_id)
+        try:
+            saml_provider = SAMLProviderConfig.objects.current_set().get(slug=enterprise_customer_idp.provider_id)
```

Comment:
sorry not familiar with current_set, what does it do?

Reply:
https://github.com/edx/django-config-models/blob/master/config_models/models.py#L33 is where it's defined.
Basically just limits the result set for you. The alternative is a reverse sort and an array limiter, ie `order_by('-id')[:1]` However, this was not functioning very well with filtering and can have some inconsistent behaviour, so glad this shortcut function exists.

Reply:
interesting, I guess this is a django specific pattern I have not seen anywhere else, hence I am still yet to learn this. hopefully this will be clearer to me once I read that new book!